routes/adminManageUser.js

```diff
@@ -0,0 +1,99 @@
+// Import required libraries and modules
+const express = require('express');
+const router = express.Router();
+const User = require('../models/user'); // User model
+const bcrypt = require('bcrypt')
+
+router.get('/', async (req, res) =>{
+  res.render('admin/adminUserMgmt')
+})
+// API route for getting all users
+router.get('/viewUser', async (req, res) => {
+  try {
+    const users = await User.find();
+    res.json(users);
+  } catch (err) {
+    res.status(500).json({ message: err.message });
+  }
+})
+
+// API route for getting a single user by ID
+router.get('/viewUser/:id', getUser, (req, res) => {
+  res.json(res.user);
+})
+
+// API route for creating a new user
+router.get('/addUser', async (req,res) => {
+  res.render('admin/addUser')
+})
+
+router.post('/addUser', async (req, res) => {
+  const hashedPass = await bcrypt.hash(req.body.password,10)
```

Comment:
does frontend have null check for password? if not add null check for password before hashing